### ISSUE TITLE
Zoom & Lightbox Not Present for Variation Images if no Main Product Image Set

### ISSUE THREAD

**Issue by jbeales** [woocommerce/woocommerce#27247]
**Prerequisites (mark completed items with an [x]):**
- [x ] I have have carried out troubleshooting steps and I believe I have found a bug.
- [ x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

**Describe the bug**
Zoom and Lightbox functionality is missing from Variation images if no main Product Image is set.

**Expected behavior**
I expect that when an image for a Variation is displayed I will be able to zoom it or view it in a lightbox, (assuming that those features are enabled on the current site).

**Actual behavior**
When selecting a Variation its image appears as normal, but it does not zoom when hovered, and there is no way to open it in a lightbox.

**Steps to reproduce the bug (We need to be able to reproduce the bug in order to fix it.)**
Steps to reproduce the bug:
1. Create a new Variable product
2. Add at least 2 variations.
3. Add an image to at least 1 variation.
4. Save the product.
5. View the front-end page for the product.
6. Select one of the variations with an image.
7. Observe that the image for the variation appears as expected.
8. Attempt to zoom the image, (hovering it on a desktop), or open it in a lightbox, (usually clicking a magnifying glass - but the magnifying glass isn't present).

**Screenshots**
Ignoring that the image is not red - this is the correct image for the "red" variation in my test store. Note that there is no magnifying glass to zoom, (this is the Storefront theme). Hovering the image with the mouse does nothing either, (it should zoom).
<img width="777" alt="Screen Shot 2020-08-06 at 8 33 36 AM" src="https://user-images.githubusercontent.com/104935/89532327-a11d5180-d7bf-11ea-8101-649f1da9ac62.png">

**Isolating the problem (mark completed items with an [x]):**
- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/). - tested with Storefront
- [x] I can reproduce this bug consistently using the steps above.

**Other Information:**
I believe this is related to #27222. It seems like without a main Product Image none of the gallery functionality gets initialized, so none of the gallery images are displayed as in #27222, and there is no Zoom/Lightbox for any Variation images that are present, (this issue).

**WordPress Environment**
<details>

```
`
### WordPress Environment ###

WordPress address (URL): http://wptest.loc
Site address (URL): http://wptest.loc
WC Version: 4.3.1
REST API Version: ✔ 1.0.10
WC Blocks Version: ✔ 2.7.2
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.3.1
Log Directory Writable: ✔
WP Version: 5.4.2
WP Multisite: –
WP Memory Limit: 3 GB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache/2.4.41 (Unix) OpenSSL/1.1.1d
PHP Version: 7.3.10
PHP Post Max Size: 8 MB
PHP Time Limit: 300
PHP Max Input Vars: 1000
cURL Version: 7.66.0
OpenSSL/1.1.1d

SUHOSIN Installed: –
MySQL Version: 5.7.27
Max Upload Size: 2 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.3.1
WC Database Prefix: wp_
Total Database Size: 4.64MB
Database Data Size: 3.11MB
Database Index Size: 1.53MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.02MB + Index: 0.11MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_options: Data: 2.13MB + Index: 0.06MB + Engine InnoDB
wp_postmeta: Data: 0.08MB + Index: 0.03MB + Engine InnoDB
wp_posts: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_bookings_availability: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_booking_relationships: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_reserved_stock: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wpsc_also_bought: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_cart_contents: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_cart_item_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_checkout_forms: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wpsc_claimed_stock: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_coupon_codes: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_currency_list: Data: 0.02MB + Index: 0.01MB + Engine MyISAM
wp_wpsc_download_status: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_meta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wpsc_product_rating: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_purchase_logs: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_purchase_meta: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_region_tax: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_submited_form_data: Data: 0.00MB + Index: 0.00MB + Engine MyISAM
wp_wpsc_visitors: Data: 0.00MB + Index: 0.01MB + Engine MyISAM
wp_wpsc_visitor_meta: Data: 0.00MB + Index: 0.01MB + Engine MyISAM

### Post Type Counts ###

attachment: 6
flamingo_contact: 2
page: 12
post: 4
product: 12
product_variation: 4
revision: 7
shop_coupon: 2
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 4.3.1

### Inactive Plugins (16) ###

Better Sold Out for WPEC: by John Beales – 0.1
Bookings Helper: by WooCommerce – 1.0.2 – Installed version not tested with active version of WooCommerce 4.3.1
Contact Form 7: by Takayuki Miyoshi – 5.2
Elementor: by Elementor.com – 2.9.14
Flamingo: by Takayuki Miyoshi – 2.2
Gallery Captions for WooCommerce: by John Beales – 1.1 – Installed version not tested with active version of WooCommerce 4.3.1
JB Widget Problems Demo: by John Beales – 1.0
JB woocommerce_single_product_image_html Demo: by John Beales – 1.0 – Installed version not tested with active version of WooCommerce 4.3.1
Porto Theme - Functionality: by P-Themes – 1.7.1
Variation Swatches for WooCommerce: by Emran Ahmed – 1.0.85 – Installed version not tested with active version of WooCommerce 4.3.1
Variation Swatches for WooCommerce - Pro: by Emran Ahmed – 1.0.54 – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Additional Variation Images: by WooCommerce – 1.7.22 (update to version 1.7.23 is available) – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Bookings: by Automattic – 1.13.3 (update to version 1.15.23 is available) – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Services: by Automattic – 1.24.0 – Installed version not tested with active version of WooCommerce 4.3.1
WooCommerce Stripe Gateway: by WooCommerce – 4.5.0 – Installed version not tested with active version of WooCommerce 4.3.1
Woo Description on Radio Buttons demo: by John Beales – 1.0

### Settings ###

API Enabled: ✔
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: booking (booking)
external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #17 - /shop/
Cart: #18 - /cart/
Checkout: #19 - /checkout/
My account: #20 - /my-account/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Storefront
Version: 2.5.8
Author URL: https://woocommerce.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: –

### Action Scheduler ###

Complete: 7
Oldest: 2020-07-15 18:38:09 +0000
Newest: 2020-08-04 13:44:04 +0000

`
```

</details>

**Comment by juliaamosova** [comment]
Hi @jbeales,

Thank you for taking the time to report this bug, we really appreciate your help. I can reproduce it on my end. 

Product with an image added to only one of the variations (missing the magnifying icon):

![icon1](https://user-images.githubusercontent.com/19143190/89580138-e44ce400-d802-11ea-86a2-a13dc30736ce.jpg)

Expected behavior - there should be a magnifying icon as shown below:

![icon2](https://user-images.githubusercontent.com/19143190/89580151-e7e06b00-d802-11ea-8f39-86c3227a29be.jpg)

This seems like not a major issue yet it affects basic use of WooCommerce which is why I am assigning this a `high` priority to fix.

**Comment by adammascherin** [comment]
When will this be fixed? I'm running a site with primarily variable products with no parent product images. I saw that the commit was reverted bc it didn't solve the problem everywhere. Is anyone following up with this?

**Comment by erizonmacani** [comment]
Hello guys.

This seems like a significant issue, so I want to share my brief inputs.

I had this same issue, which somehow indirectly got resolves by the following code:

`add_filter( 'woocommerce_hide_invisible_variations', '__return_false' );`

The function of this code was to allow product variants to show even if there is not price (normally woocommerce will hide a variant if it has no price)

I wanted to dig in to see why this fixes the issue, but I don't have the time, so the least I could do is share this input.

Good luck with finding the solution!

**Comment by sunyatasattva** [comment]
Removing `priority: high` label. We will address this as related to #27222, but:

- it's a relatively edge case
- has a workaround
- belongs to classic front end

**Comment by Aljullu** [comment]
@jbeales once again, thanks for filling this issue! We have resolved it in https://github.com/woocommerce/woocommerce/pull/47714, so the fix should be included in the WooCommerce 9.1 release, expected for July 9th.
